packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -0,0 +1,236 @@
+// @flow
+import moment from "moment";
+import * as React from "react";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {Body} from "@khanacademy/wonder-blocks-typography";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Color from "@khanacademy/wonder-blocks-color";
+import {SingleSelect, OptionItem} from "@khanacademy/wonder-blocks-dropdown";
+
+type Props = {|
+    /**
+     * The default value to populate the birthdate with. Should be in the
+     * format: YYYY-MM-DD (e.g. 2021-05-26). It's only used to populate the
+     * initial value as this is an uncontrolled component.
+     */
+    defaultValue?: string,
+
+    /**
+     * Listen for changes to the birthdate. Could be a string in the YYYY-MM-DD
+     * format or `null`.
+     */
+    onChange: (date: ?string) => mixed,
+|};
+
+type State = {|
+    month: string | null,
+    day: string | null,
+    year: string | null,
+    error: string | null,
+|};
+
+// Flow doesn't know about the getYear property on Date for some reason!
+// $FlowFixMe[prop-missing]
+const CUR_YEAR = new Date().getYear() + 1900;
+// TODO(WB-1200.2): Add labels prop to handle translations.
+const BIRTHDAY_ERROR = "Please select a valid birthdate.";
```

Comment:
NOTE: These hardcoded strings will be changed to use a `labels` prop in the next PR (same applies to month, day, year).